compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -832,6 +843,11 @@ private Toml getToml(TomlTableNode toolNode, String key) {
         return new Toml(optionsNode);
     }
 
+    private boolean providedPlatformDependencyIsValid(String artifactId, String groupId, String version) {
+        return artifactId != null && !artifactId.isEmpty() && groupId != null && !groupId.isEmpty()
+                && version != null && !version.isEmpty();
```

Comment:
```suggestion
         return  !"".equals(artifactId) && !"".equals(groupId)  && !groupId.isEmpty() && !"".equals(version)
```

can't we do this? It checks `null` too.

Reply:
`!"".equals(artifactId)` does not work when `artifactId` is null.